plaso/cli/image_export_tool.py

```diff
@@ -223,66 +226,68 @@ def _ExtractDataStream(
     display_name = path_helper.PathHelper.GetDisplayNameForPathSpec(
         file_entry.path_spec)
 
-    if skip_duplicates:
-      try:
-        digest = self._CalculateDigestHash(file_entry, data_stream_name)
-      except (IOError, dfvfs_errors.BackEndError) as exception:
-        output_writer.Write((
-            '[skipping] unable to read content of file entry: {0:s} '
-            'with error: {1!s}\n').format(display_name, exception))
-        return
+    try:
+      digest = self._CalculateDigestHash(file_entry, data_stream_name)
+    except (IOError, dfvfs_errors.BackEndError) as exception:
+      logger.error((
+          '[skipping] unable to read content of file entry: {0:s} '
+          'with error: {1!s}').format(display_name, exception))
+      return
+
+    target_directory, target_filename = self._CreateSanitizedDestination(
+        file_entry, file_entry.path_spec, data_stream_name, destination_path)
+
+    target_path = os.path.join(target_directory, target_filename)
 
+    self._report_output[digest].append(target_path.replace(
```

Comment:
Please break this into multiple statements, one doing the path replacement, and another doing the addition the dict.

Reply:
On line 246 you're checking for digest being populated, but you're also using it here. What's supposed to happen if digest is None? Can _CalculateDigestHash return None, or is the check on 246 redundant?

Reply:
_CalculateDigestHash can return None, if the digest cannot be determined. This check (if not digest) should actually happen slightly before, thanks for spotting that.